src/Quantum/Quantum.jl

```diff
@@ -43,69 +39,96 @@ function plug end
 plug(::T) where {T<:TensorNetwork} = plug(T)
 plug(::Type{T}) where {T<:TensorNetwork} = plug(ansatz(T))
 
-sites(tn::TensorNetwork; dir::Symbol = :all) = sites(tn, dir)
-@valsplit 2 sites(tn::TensorNetwork, dir::Symbol) = throw(MethodError(sites, "dir=$dir not recognized"))
-sites(tn::TensorNetwork, ::Val{:all}) = unique(first.(keys(tn[:plug])))
-sites(tn::TensorNetwork, ::Val{:in}) = first.(filter(==(:in) ∘ last, keys(tn[:plug])))
-sites(tn::TensorNetwork, ::Val{:out}) = first.(filter(==(:out) ∘ last, keys(tn[:plug])))
-
-labels(tn::TensorNetwork, ::Val{:plug}) = unique(values(tn[:plug]))
-labels(tn::TensorNetwork, ::Val{:plug}, site) = labels(tn, Val(:in), site) ∪ labels(tn, Val(:out), site)
-labels(tn::TensorNetwork, ::Val{:in}) = map(last, Iterators.filter((((_, dir), _),) -> dir === :in, tn[:plug]))
-labels(tn::TensorNetwork, ::Val{:in}, site) = tn[:plug][(site, :in)]
-labels(tn::TensorNetwork, ::Val{:out}) = map(last, Iterators.filter((((_, dir), _),) -> dir === :out, tn[:plug]))
-labels(tn::TensorNetwork, ::Val{:out}, site) = tn[:plug][(site, :out)]
+sites(tn::TensorNetwork) = collect(mapreduce(keys, ∪, tn.interlayer))
+
+labels(tn::TensorNetwork, ::Val{:plug}) = unique(Iterators.flatmap(values, tn.interlayer))
```

Comment:
```suggestion
labels(tn::TensorNetwork, ::Val{:plug}) = unique(Iterators.flatten(map(values, tn.interlayer)))
```
It seems that it can not find the `flatmap` function, maybe try `flatten(map(...))` instead?

Reply:
Ahh, it looks like it was added in Julia 1.9.

Reply:
```suggestion
labels(tn::TensorNetwork, ::Val{:plug}) = unique(Iterators.flatten(Iterators.map(values, tn.interlayer)))
```